sdk/tx-kernel/src/externs.rs

```diff
@@ -15,7 +15,7 @@ extern "C" {
 #[link(wasm_import_module = "miden:tx_kernel/note")]
 extern "C" {
     #[link_name = "get_inputs<0x0000000000000000000000000000000000000000000000000000000000000000>"]
-    pub fn extern_note_get_inputs(ptr: *mut Felt) -> usize;
+    pub fn extern_note_get_inputs(ptr: i32) -> usize;
```

Comment:
Do you know why the pointer type is causing an issue here? We should try to figure that out, since I'm surprised that this ends up being irrelevant (as both pointers and 32-bit integers should basically have the same representation when compiled to MASM). I'm guessing we must be doing something wrong during codegen if this is causing an error.

Reply:
Oh, my bad. The issue was the `Felt` instead of `i32` at https://github.com/0xPolygonMiden/compiler/blob/e661c17f4e15ed2a0ae8554cfb599a1c1d50778a/frontend-wasm/src/miden_abi/tx_kernel/note.rs#L11
Here, I changed it to `i32` for some weird reason, probably I was tired.  `*mut Felt` works just fine, I restored it.

Reply:
Good catch! Thanks!